### ISSUE TITLE
Empty Responses Issue in Output

### ISSUE THREAD

**Issue by wfr429** [JackChen-seu/Reefknot#5]
Thanks for your great work! I ran `infer.sh` as described in the README.md, but I noticed that some outputs have empty responses. Can you help me fix this issue?

```
{"image_id": "1592267", "query_prompt": "Is the bus driver on bus in this photo? Please answer yes or no.", "response": "Yes", "label": "yes", "mllm_name": "llava-v1.5-13b"}
{"image_id": "1592267", "query_prompt": "Is the bus driver off the bus in this photo? Please answer yes or no.", "response": "", "label": "no", "mllm_name": "llava-v1.5-13b"}
{"image_id": "2370275", "query_prompt": "Are the sandwich on plate in this photo? Please answer yes or no.", "response": "Yes", "label": "yes", "mllm_name": "llava-v1.5-13b"}
{"image_id": "2370275", "query_prompt": "Is the sandwich under the plate in this photo? Please answer yes or no.", "response": "", "label": "no", "mllm_name": "llava-v1.5-13b"}
{"image_id": "2340467", "query_prompt": "Is the banner on building in this photo? Please answer yes or no.", "response": "Yes", "label": "yes", "mllm_name": "llava-v1.5-13b"}
{"image_id": "2340467", "query_prompt": "Is the banner next to the building in this photo? Please answer yes or no.", "response": "Yes", "label": "no", "mllm_name": "llava-v1.5-13b"}
```

**Comment by xnumad** [comment]
I am facing the same issue, have you found any insights on this?

**Comment by xnumad** [comment]
DTC.py: When doing masking for relative_top truncation, it sets values to "-inf".
Then when doing the calibration, it calculates `float("-inf") - float("-inf")`, which is NaN.
NaN dominates over other values, which is propagated in argmax as it returns index 0. Index 0 is <unk> token in LLaVa.
This failure case stops when generating some next token; DTC usually won't trigger, so model continues normally, and usually ends generation.

Fix: Avoid NaN:
Replace "-inf" with -1e9 in https://github.com/JackChen-seu/Reefknot/blob/310ec0c468f7763440ca4ffb53f1fe00d9a3deb7/DTC.py#L161

**Comment by wfr429** [comment]
> DTC.py: When doing masking for relative_top truncation, it sets values to "-inf". Then when doing the calibration, it calculates `float("-inf") - float("-inf")`, which is NaN. NaN dominates over other values, which is propagated in argmax as it returns index 0. Index 0 is token in LLaVa. This failure case stops when generating some next token; DTC usually won't trigger, so model continues normally, and usually ends generation.
> 
> Fix: Avoid NaN: Replace "-inf" with -1e9 in
> 
> [Reefknot/DTC.py](https://github.com/JackChen-seu/Reefknot/blob/310ec0c468f7763440ca4ffb53f1fe00d9a3deb7/DTC.py#L161)
> 
> Line 161 in [310ec0c](/JackChen-seu/Reefknot/commit/310ec0c468f7763440ca4ffb53f1fe00d9a3deb7)
> 
>  base_logits_norm = base_logits_norm.masked_fill(mask, float("-inf"))

Got it🫡. I’ll give it a try. Thanks for your effort.
